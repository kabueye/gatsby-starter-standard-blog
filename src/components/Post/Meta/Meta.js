// @flow strict
import React from 'react';
import moment from 'moment';
import styles from './Meta.module.scss';
import { useSiteMetadata } from '../../../hooks';

type Props = {
  date: string
};

const Meta = ({ date }: Props) => {
  const { labelPublished, dateFormat } = useSiteMetadata();

  return (
    <div className={styles['meta']}>
      <p className={styles['meta__date']}>{labelPublished} {moment(date).format(dateFormat)}</p>
    </div>
  );
};

export default Meta;
