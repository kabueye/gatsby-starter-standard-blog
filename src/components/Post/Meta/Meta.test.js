// @flow strict
import React from 'react';
import { useStaticQuery, StaticQuery } from 'gatsby';
import renderer from 'react-test-renderer';
import Meta from './Meta';
import siteMetadata from '../../../../jest/__fixtures__/site-metadata';
import type { RenderCallback } from '../../../types';

describe('Meta', () => {
  const mockProps = {
    ...siteMetadata,
  };

  beforeEach(() => {
    StaticQuery.mockImplementationOnce(
      ({ render }: RenderCallback) => (
        render(mockProps)
      ),
      useStaticQuery.mockReturnValue(mockProps)
    );
  });

  it('renders correctly', () => {
    const props = {
      date: '2016-09-01'
    };

    const tree = renderer.create(<Meta {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
