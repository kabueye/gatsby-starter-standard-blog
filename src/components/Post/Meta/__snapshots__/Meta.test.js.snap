// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meta renders correctly 1`] = `
<div
  className="meta"
>
  <p
    className="meta__date"
  >
    Published on
     
    September 1, 2016
  </p>
</div>
`;
